test/Tester/ActivationServices/IStorageActivationService.cs

```diff
@@ -0,0 +1,78 @@
+﻿
+using Orleans.Runtime;
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Orleans;
+
+namespace Tester
+{
+    public interface IStorageActivationService<TState>
+    {
+        string Name { get; }
+
+        TState State { get; set; }
+
+        Task SaveAsync();
+
+        string GetExtendedInfo();
+    }
+
+    [AttributeUsage(AttributeTargets.Parameter)]
+    public class StorageActivationServiceAttribute : Attribute
+    {
+        public string StateName { get; }
+        public string StorageProviderName { get; }
+
+        public StorageActivationServiceAttribute(string storageProviderName = null, string stateName = null)
+        {
+            this.StorageProviderName = storageProviderName;
+            this.StateName = stateName;
+        }
+    }
+
+    public class AttributedStorageActivationService<TState> : IStorageActivationService<TState>
+    {
+        private readonly IStorageActivationService<TState> storageActivationService;
+
+        public AttributedStorageActivationService(StorageActivationServiceFactory<TState> factory, IGrainActivationContext context)
+        {
+            var parameter = ActivationServiceUtilities.BindToConstructorParameter(context, typeof(IStorageActivationService<TState>));
```

Comment:
I see. if it is not shared by all activation services, then I withdraw my comments. 

And I second @sergeybykov 's comment, maybe more high description will help review. For example, what's the approach of using BindToConstructorParameters in an activation service. How does it wire up internally. And what's other approaches and what's their difference.